### PR TITLE
added instructions to cross compile aci-cni for arm

### PR DESCRIPTION
**Pull request by anmol372**
and then build arm-specific cni container images.

preview doc: https://github.com/noironetworks/acc-provision/blob/arm64-readme/doc/features/build-arm-images.md

### CODE REVIEW

**Comment by anmol372** [comment]
should we list and link all the Dockerfiles that the customer would need to update and build as well?

File: doc/features/build-arm-images.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+###  Define Container Image Registry and Tag:
+    
+We will first define the image build registry and image tag, this will build all container images as
+`$IMAGE_BUILD_REGISTRY/<Image_Name>:IMAGE_BUILD_TAG`
+    
+    ```
+    export IMAGE_BUILD_REGISTRY="my_registry"
+    export IMAGE_BUILD_TAG="v0.1"
+    ```
```
Comment (snaiksat) [inline doc/features/build-arm-images.md:108]: Add:

Replace "my_registry" and "v0.1" with the relevant registry name and tag in your environment.

File: doc/features/build-arm-images.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    We will build  opflex images with the script `docker/travis/build-opflex-travis.sh`.
+    ```
+    docker/travis/build-opflex-travis.sh $IMAGE_BUILD_REGISTRY $IMAGE_BUILD_TAG
+    ```
+
+    This script can take three parameters:
+    1. Image Registry
+    2. Image Tag
+    3. Build args: We can provide additional build arguments for docker buildx build commands, for example if you are using an http proxy you may use the script as:
+    `docker/travis/build-opflex-travis.sh $IMAGE_BUILD_REGISTRY $IMAGE_BUILD_TAG "--build-arg https_proxy=${HTTP_PROXY} --build-arg no_proxy=${NO_PROXY}"`
```
Comment (snaiksat) [inline doc/features/build-arm-images.md:131]: move the command to a new line

File: doc/features/build-arm-images.md [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+`./build_openvswitch.sh $IMAGE_BUILD_REGISTRY $IMAGE_BUILD_TAG "--build-arg https_proxy=${HTTP_PROXY} --build-arg no_proxy=${NO_PROXY}"`
+
+The above steps will build two containers:
+```
+1. $IMAGE_BUILD_REGISTRY/openvswitch-base:$IMAGE_BUILD_TAG
+2. $IMAGE_BUILD_REGISTRY/openvswitch:$IMAGE_BUILD_TAG
+```
+
+###  Build ARM-specific ACI-CNI Container Images 
+We will use Docker Buildx to create multi-architecture images: `docker buildx build --no-cache -t <tag>  --platform linux/arm64 -f Dockerfile --push .`
```
Comment (snaiksat) [inline doc/features/build-arm-images.md:183]: move the command to a new line

File: doc/features/build-arm-images.md [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+
+**1. Build the Containers:**
+
+```
+docker buildx build --no-cache --platform linux/arm64 -t ${IMAGE_BUILD_REGISTRY}/aci-containers-controller:${IMAGE_BUILD_TAG} --file=docker/travis/Dockerfile-controller .
+docker buildx build --no-cache --platform linux/arm64 --target without-ovscni -t ${IMAGE_BUILD_REGISTRY}/aci-containers-host:${IMAGE_BUILD_TAG} --file=docker/travis/Dockerfile-host .
+docker buildx build --no-cache --platform linux/arm64 -t ${IMAGE_BUILD_REGISTRY}/aci-containers-operator:${IMAGE_BUILD_TAG} --file=docker/travis/Dockerfile-operator .
```
Comment (snaiksat) [inline doc/features/build-arm-images.md:190]: Either acc-provision-operator needs to be added, or the configuration to disable use of acc-provision-operator should be mentioned: https://github.com/noironetworks/acc-provision/blob/master/doc/features/acc-provision-operator-exclude.md

File: doc/features/build-arm-images.md [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+Run the following command to check the images:
+```
+docker images
+REPOSITORY                               TAG                       IMAGE ID       CREATED         SIZE
+my_registry/aci-containers-operator      v0.1-test                 2bccbe38a7fd   7 hours ago     589MB
+my_registry/aci-containers-controller    v0.1-test                 aa8815e53b15   7 hours ago     638MB
+my_registry/aci-containers-host          v0.1-test                 1a2300555aee   7 hours ago     727MB
+my_registry/openvswitch                  v0.1-test                 1365a40ef73f   7 hours ago     681MB
+my_registry/openvswitch-base             v0.1-test                 8f36e0df45cf   7 hours ago     1.33GB
+```
```
Comment (snaiksat) [inline doc/features/build-arm-images.md:204]: we need to include the opflex and acc-provision-operator images in this list

File: doc/features/build-arm-images.md [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+    ```
+    git apply ./arm64-aci.patch
+    ```
+4. **Build ACI Containers binaries**
+    ```
+    make clean-dist-static all-static
+    ``` 
+5. **Copy Iptables binaries**
+
+    Once the binaries have been built we need to copy some additional files to the dist-static directory in aci-containers. We will use `docker/copy_iptables.sh` script to copy iptables binaries and libraries from opflex-build-base container built in the arlier step into dist-static to be used by consumers of iptables like the host agent container.
```
Comment (snaiksat) [inline doc/features/build-arm-images.md:157]: > arlier

typo: earlier

**Review by snaiksat — CHANGES_REQUESTED** [review]

File: doc/features/build-arm-images.md (hunk 4 above)
Comment (anmol372) [reply doc/features/build-arm-images.md:194]: Added the following, will add acc-provision-operator in a follow up pr.
`
TODO: Add steps to build acc-provision-operator.
Please follow steps to [exclude acc provision operator](https://github.com/noironetworks/acc-provision/blob/master/doc/features/acc-provision-operator-exclude.md) for now.
`

File: doc/features/build-arm-images.md (hunk 5 above)
Comment (anmol372) [reply doc/features/build-arm-images.md:213]: included opflex, will include acc-provision-operator in the follow up PR.

Comment (anmol372) [reply doc/features/build-arm-images.md:226]: added info to build acc-provision-operator as well.

**Review by snaiksat — APPROVED** [review]

**Pull request merged**